server/sonar-web/src/main/js/apps/custom-metrics/components/List.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import { sortBy } from 'lodash';
+import DeleteButton from './DeleteButton';
+import EditButton from './EditButton';
+import { MetricProps } from './Form';
+import { Metric } from '../../../app/types';
+import ActionsDropdown, {
+  ActionsDropdownDivider
+} from '../../../components/controls/ActionsDropdown';
+import { translate } from '../../../helpers/l10n';
+
+interface Props {
+  domains?: string[];
+  metrics: Metric[];
+  onDelete: (metricKey: string) => Promise<void>;
+  onEdit: (data: { id: string } & MetricProps) => Promise<void>;
+  types?: string[];
+}
+
+export default function List({ domains, metrics, onDelete, onEdit, types }: Props) {
+  return (
+    <div id="custom-metrics-list" className="boxed-group boxed-group-inner">
+      <table className="data zebra zebra-hover">
+        <tbody>
+          {sortBy(metrics, metric => metric.name.toLowerCase()).map(metric => (
```

Comment:
I just thought about this: we should display a message if there is no custom metrics yet.